app/views/purchases/_buttons.html.erb

```diff
@@ -0,0 +1,13 @@
+<%= form_with url: {action: "limit"} do |f| %>
+  <div class="field is-grouped line-spacing-bottom limit-button-spacing">
+    <div class="control">
+      <% [5, 10, 20, 50, 100].each do |value| %>
+        <% if value == plimit %>
+          <%= f.submit value, value: value, id: "button_limit_#{value}", class: 'button is-dark' %>
+        <% else %>
+          <%= f.submit value, value: value, id: "button_limit_#{value}", class: 'button' %>
```

Comment:
I'd recommend to determine whether the `is-dark` class should be added with a helper function.

`rails generate helper purchase`

Helpers are for view concerns so it's a good usecase for one  
Something like this:

```ruby
def active_limit(val, limit)
   'is-dark' if val == limit
end
```

`<%= f.submit value, value: value, id: "button_limit_#{value}", class: active_limit(value, @plimit) %>`

Reply:
Oh, hadn't used helpers before - that tidies things up nicely. Just spent five minutes banging my head against the wall because I was tiredly writing the function in the spec/helpers, not app/helpers >_<
Tweaked it slightly to return an empty string rather than nil if it doesn't match.